pyclean/modern.py

```diff
@@ -90,6 +90,50 @@ def configure(self, args):
 Runner = CleanupRunner()
 
 
+def normalize_path(pattern: str) -> str:
+    """
+    Normalize path separators in a pattern for cross-platform support.
+
+    On Windows, both forward slash and backslash are valid path separators.
+    On Unix/Posix, only forward slash is valid (backslash can be part of filename).
+    """
+    return pattern.replace(os.sep, os.altsep or os.sep)
+
+
+def should_ignore(path: Path, ignore_patterns: list) -> bool:
```

Comment:
Shouldn't the type annotation be more specific, like this?

```suggestion
def should_ignore(path: Path, ignore_patterns: list[str]) -> bool:
```